src/main/java/io/vertx/core/net/TrafficShapingOptions.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2011-2019 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.core.net;
+
+import io.netty.util.internal.ObjectUtil;
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.codegen.annotations.Unstable;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Options describing how {@link io.netty.handler.traffic.GlobalTrafficShapingHandler} will handle traffic shaping.
+ */
+@Unstable
+@DataObject(generateConverter = true, publicConverter = false)
+public class TrafficShapingOptions {
+  /*
+   * Default inbound bandwidth limit in bytes/sec = 0 (0 implies unthrottled)
+   */
+  public static final long DEFAULT_INBOUND_GLOBAL_BANDWIDTH_LIMIT = 0;
+
+  /**
+   * Default outbound bandwidth limit in bytes/sec = 0 (0 implies unthrottled)
+   */
+  public static final long DEFAULT_OUTBOUND_GLOBAL_BANDWIDTH_LIMIT = 0;
+
+  private long inboundGlobalBandwidth;
+  private long outboundGlobalBandwidth;
+  private long peakOutboundGlobalBandwidth;
+  private long maxDelayToWaitTime;
```

Comment:
we need a `maxDelayToWaitTimeUnit` as well that is a `TimeUnit`, e.g we do have in `TCPSSLOptions` `idleTimeout` and `idleTimeoutUnit`